backend/app/services/knowledge/knowledge_service.py

```
@@ -420,12 +420,28 @@ def list_knowledge_bases(
 
-            # Single query to get personal, team, organization, shared, and bound knowledge bases
-            # Personal: user_id matches and namespace is "default"
-            # Team: namespace is in accessible_groups
+            # External resolver: returns additional KB IDs the user can access
+            # via extension rules (e.g. department / employee bindings).
+            # Errors are caught so a faulty extension cannot break the core listing path.
+            from app.services.readers.kb_permissions import kb_permission_resolver
+
+            try:
+                ext_kb_ids = kb_permission_resolver.get_accessible_kb_ids(db, user_id)
+            except Exception as e:
```

Ruff (0.15.11)

[warning] Do not catch blind exception: `Exception`

(BLE001)